### PR TITLE
frei0r: Add option for cairo and opencv

### PR DESCRIPTION
**Pull request by s172262**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

I'm sorry that this is re-open of #710.

~~Using **cmake** because `configure` couldn't find out `cairo`.~~
Using **configure** because there is no switch for **disable** `cairo` in CMakeLists.txt.

### CODE REVIEW

File: Formula/frei0r.rb [hunk 1]
```diff
@@ -13,10 +14,13 @@ class Frei0r < Formula
...
 
   depends_on "autoconf" => :build
+  depends_on "pkg-config" => :build
+  depends_on "cairo" => :optional
+  depends_on "homebrew/science/opencv" => :optional
 
   def install
-    system "./configure", "--disable-dependency-tracking",
-                          "--prefix=#{prefix}"
+    ENV["CAIRO_CFLAGS"] = "-I#{Formula["cairo"].opt_include}/cairo" if build.with? "cairo"
```
Comment (DomT4) [inline Formula/frei0r.rb]: Surprised you need this?

File: Formula/frei0r.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@ class Frei0r < Formula
   homepage "http://frei0r.dyne.org"
   url "https://files.dyne.org/frei0r/releases/frei0r-plugins-1.4.tar.gz"
   sha256 "8470fcabde9f341b729be3be16385ffc8383d6f3328213907a43851b6e83be57"
+  revision 1
```
Comment (DomT4) [inline Formula/frei0r.rb]: Don't need the revision bump here as you're not changing the existing build.

File: Formula/frei0r.rb (hunk 1 above)
Comment (s172262) [inline Formula/frei0r.rb]: Yes, CC can't find out <cairo.h> without `CAIRO_CFLAGS` because it wont see /cairo directory.

**Comment by UniqMartin** [comment]
I'm really sorry @s172262. While squashing and cleaning up your commits locally, I managed to accidentally remove you as the commit author. 😢

Merged in 016e5041aab9b9db10f32b44abbafd16ca135b9f. Thank you for this contribution to Homebrew, @s172262! :tada:

**Pull request closed without merging**